crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,223 @@
+use anyhow::Result;
+use async_trait::async_trait;
+use swc_core::ecma::ast::Program;
+use turbo_tasks::primitives::StringVc;
+use turbo_tasks_fs::FileSystemPathVc;
+use turbopack_core::issue::{Issue, IssueSeverity, IssueSeverityVc, IssueVc};
+use turbopack_ecmascript::{CustomTransformer, TransformContext};
+
+/// A wrapper around a SWC's ecma transform wasm plugin module bytes, allowing
+/// it to operate with the turbo_task caching requirements.
+/// Internally this contains a `CompiledPluginModuleBytes`, which points to the
+/// compiled, serialized wasmer::Module instead of raw file bytes to reduce the
+/// cost of the compilation.
+#[turbo_tasks::value(
+    transparent,
+    serialization = "none",
+    eq = "manual",
+    into = "new",
+    cell = "new"
+)]
+pub struct SwcPluginModule(
+    #[turbo_tasks(trace_ignore)]
+    #[cfg(feature = "swc_ecma_transform_plugin")]
+    pub swc_core::plugin_runner::plugin_module_bytes::CompiledPluginModuleBytes,
+    // Dummy field to avoid turbo_tasks macro complains about empty struct.
+    // This is due to we can't import CompiledPluginModuleBytes by default, it should be only
+    // available for the target / platforms can support swc plugins (which can build wasmer)
```

Comment:
```suggestion
    // Dummy field to avoid turbo_tasks macro complaining about empty struct.
    // This is because we can't import CompiledPluginModuleBytes by default, it should be only
    // available for the target / platforms that support swc plugins (which can build wasmer)
```